sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -534,6 +535,14 @@ object ViewHelper extends SQLConfHelper with Logging {
     }
   }
 
+  def verifyAutoGeneratedAliasesNotExists(child: LogicalPlan, name: TableIdentifier): Unit = {
+    child.expressions.foreach {
```

Comment:
We should check `child.output`, instead of expressions. e.g.
```
SELECT * FROM (SELECT a + b FROM t)
```

The root plan here is a `Project` with `AttributeReference` only, and the check here can't catch it.